// lmsproject/src/pages/user/courses.jsx
export const courses = [
  {
    id: 1,
    title: "MERN Stack Mastery",
    instructor: "John Doe",
    price: "119 $",
    image:
      "https://images.unsplash.com/photo-1518770660439-4636190af475?q=80&w=1200&auto=format&fit=crop",
  },

  {
    id: 2,
    title: "DSA For Placements",
    instructor: "Sarah Lee",
    price: "199 $",
    image:
      "https://images.unsplash.com/photo-1551434678-e076c223a692?q=80&w=1200&auto=format&fit=crop",
  },

  {
    id: 3,
    title: "Frontend System Design",
    instructor: "Alex Smith",
    price: "349 $",
    image:
      "https://images.unsplash.com/photo-1498050108023-c5249f4df085?q=80&w=1200&auto=format&fit=crop",
  },

  {
    id: 4,
    title: "Blockchain Development",
    instructor: "Jhon joseph",
    price: "599 $",
    image: "https://images.unsplash.com/photo-1510511459019-5dda7724fd87?q=80&w=1200&auto=format&fit=crop"
  },
  {
    id: 5,
    title: "Advance DSA",
    instructor: "AKS",
    price: "299 $",
    image: "https://images.unsplash.com/photo-1519389950473-47ba0277781c?q=80&w=1200&auto=format&fit=crop"
  },
  {
    id: 6,
    title: "RUST",
    instructor: "Fisher Nolan",
    price: "399 $",
    image: "https://images.unsplash.com/photo-1555617117-08fe6d2b3e9d?q=80&w=1200&auto=format&fit=crop"
  }
];
